Helper/Data.php

```diff
@@ -1184,9 +1192,18 @@ public function buildRequestHeaders($payment = null)
             HeaderDataBuilderInterface::MERCHANT_APPLICATION_VERSION  => $this->getModuleVersion()
         ];
 
-        if(isset($payment) && !is_null($payment->getAdditionalInformation(HeaderDataBuilderInterface::ADDITIONAL_DATA_FRONTEND_TYPE_KEY))) {
-            $headers[HeaderDataBuilderInterface::EXTERNAL_PLATFORM_FRONTEND_TYPE] =
-                $payment->getAdditionalInformation(HeaderDataBuilderInterface::ADDITIONAL_DATA_FRONTEND_TYPE_KEY);
+        if (isset($payment)) {
+            $frontendType = $payment->getAdditionalInformation(HeaderDataBuilderInterface::ADDITIONAL_DATA_FRONTEND_TYPE_KEY);
+            if (is_null($frontendType)) {
+                // Check the request URI
+                $requestUri = $this->request->getPathInfo();
+                if (str_contains($requestUri, '/graphql')) {
```

Comment:
This logical comparison might create false-positives. Let's discuss it separately.